app/controllers/TicketBooking.scala

```diff
@@ -0,0 +1,204 @@
+package controllers
+
+import models._
+import play.api.data.Forms._
+import play.api.data._
+import play.api.mvc._
+import play.filters.csrf.CSRF
+
+import java.time.LocalDateTime
+import javax.inject._
+import scala.util.{Failure, Success, Try}
+
+@Singleton
+class TicketBooking @Inject() (cc: MessagesControllerComponents) extends MessagesAbstractController(cc) {
+
+  private val availableScreenings: List[Screening] = DefaultScreenings.getAvailableScreenings
+  private var availableScreeningsFiltered: List[Screening] = Nil
+  private var chosenScreeningOption: Option[Screening] = None
+  private var chosenSeats: Seq[(Int, Int)] = Nil
+  private var chosenTickets: Seq[Ticket] = Nil
+
+  val customerDataForm: Form[Customer] = Form(mapping(
+    "name" -> text(3),
+    "surname" -> text(3)
+  )(Customer.apply)(Customer.unapply))
+
+  def ticketBookingInit: Action[AnyContent] = Action { implicit request =>
+    val csrfToken = CSRF.getToken.get.value
+    Ok(views.html.ticketBookingInit()).withSession("csrfToken" -> csrfToken)
+  }
+
+  def chooseDateAndTime: Action[AnyContent] = Action { implicit request =>
+    val postVals = request.body.asFormUrlEncoded
+    postVals.map { args =>
+      if (args("dateTime").head.nonEmpty) {
+        val chosenDateTime = args("dateTime").head
+        println(chosenDateTime)
+        Redirect(routes.TicketBooking.availableMovies).withSession(request.session.+("dateTime" -> chosenDateTime))
+      } else {
+        println("empty date time")
```

Comment:
what will you use instead of `println` in real/production app?

Reply:
I'm sorry - this was a println() for testing purposes and I forgot to delete it before pushing the code. What really matters for the user is:
edirect(routes.TicketBooking.ticketBookingInit).flashing("error" -> "choose date and time!")
So, the customer sees a flashing message saying "choose date and time!"